Add doc for troubleshooting server restarts

kshepard opened the pull request:
## Overview

When servers are rebooted, there is generally some manual intervention needed in order to get the application functioning again. This documents a basic troubleshooting approach.

### Checklist

- [x] PR has a descriptive enough title to be useful in changelogs

### Notes

Also fixed a couple unrelated typos within the document

## Testing Instructions

 * No testing needed, just read through the document and make sure it looks accurate.

Closes [PT165724357](https://www.pivotaltracker.com/story/show/165724357)

rmartz commented on doc/system-administration.md:
```diff
@@ -179,10 +179,91 @@ There should be an entry for `certbot renew`. That command may be run manually a
+In an environment without Security Groups, a firewall must be configured on each machine to allow such restrictions. The firewall that comes configured with DRIVER by default is called `ufw`. The following is a set of general troubleshooting steps that may be performed when the servers are rebooted and the application no longer seems to behaving properly.
+
+
+### 1. Ensure ufw is running
```
I believe UFW as a name should be capitalized
```suggestion
### 1. Ensure UFW is running
```

rmartz commented on doc/system-administration.md:
````diff
@@ -179,10 +179,91 @@ There should be an entry for `certbot renew`. That command may be run manually a
+```
+sudo service redis-server restart
+```
+There may also be problems with restarting `driver-app` in the above step if Redis is not behaving properly. If the App machine is showing Redis-related errors in `/var/log/upstart/driver-app.log`, then be sure to restart Redis before attempting to restart `driver-app`.
````
> [...] then be sure to restart Redis before attempting to restart `driver-app`.

Can you think of any way to check if we'd experience Redis related errors before bringing `driver-app` back up? It'd be good if we could insert something about detecting/fixing before that step if we can (or else have an alternate path here for people who already brought `driver-app` up that might be blocking a Redis restart), that way a person going through these step by step doesn't run into an unpleasant surprise here

rmartz commented on doc/system-administration.md:
````diff
@@ -179,10 +179,91 @@ There should be an entry for `certbot renew`. That command may be run manually a
+Redis needs to have open permissions due to its usage via Windshaft. It is therefore paramount that network traffic is locked down via `ufw` or Security Groups. If the machine is running without a firewall in place for an extended period of time, there is a chance an automated system will be able to write data to it and cause a corrupt state. Once it's verified that `ufw` is running, check for an ...
+```
+sudo service redis-server status
+```
````
We should specify that these commands are run on the Database machine

rmartz commented on doc/system-administration.md:
```diff
@@ -179,10 +179,91 @@ There should be an entry for `certbot renew`. That command may be run manually a
+
+### 5. Verify the application is functioning
+
+All services should now be running, reporting valid status, and showing good log messages: ufw and Docker are running on each machine, `driver-app` is running on the App machine, `driver-celery` is running on the Celery machine, and Redis is running on the Database machine.
```
Similar comment about UFW used as a name:
```suggestion
All services should now be running, reporting valid status, and showing good log messages: UFW and Docker are running on each machine, `driver-app` is running on the App machine, `driver-celery` is running on the Celery machine, and Redis is running on the Database machine.
```

rmartz commented on doc/system-administration.md:
```diff
@@ -179,10 +179,91 @@ There should be an entry for `certbot renew`. That command may be run manually a
+
+All services should now be running, reporting valid status, and showing good log messages: ufw and Docker are running on each machine, `driver-app` is running on the App machine, `driver-celery` is running on the Celery machine, and Redis is running on the Database machine.
+
+Log into the web application, view the dashboard, the record list page, and the map page to ensure records are displaying properly in each of the pages. Then attempt to perform a CSV export on the map page. If it completes successfully, then the app is back in a functional state. If it times out, log into the Celery machine to see if jobs are processing. View tail the log file at: `/var/log/upsta ...
```
"View tail the log file" reads a little awkwardly to me, how about 
> Follow the log file with: `tail -f /var/log/upstart/driver-celery.log`
```suggestion
Log into the web application, view the Dashboard, the Record List page, and the Map page to ensure records are displaying properly in each of the pages. Then attempt to perform a CSV export on the Map page. If it completes successfully, then the app is back in a functional state. If it times out, log into the Celery machine to see if jobs are processing. Follow the log file with:
\```
tail -f /var/log/upstart/driver-celery.log
\```
Hopefully there are some jobs processing, demonstrated by recent log messages such as:
```
(Not sure how to embed triple-backticks inside a triple-backtick...)

kshepard commented:
Thanks for the comments, I've incorporated them. I added a note about checking for Redis errors while in the `driver-app` step, since I believe there will be some visible Redis errors in the log at that point.
